File: 104/maximum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function (root) {
    let maxn = -1;

    function dfs(root, depth) {
        if (!root) {
            maxn = Math.max(maxn, depth);
            return;
        }
        dfs(root.left, depth + 1);
        dfs(root.right, depth + 1);
    }

    dfs(root, 0);
    return maxn;
};

File: 104/MaximumDepthOfBinaryTree.scala
object MaximumDepthOfBinaryTree {
  def maxDepth(root: TreeNode): Int =
    if (root == null) 0
    else {
      val (l, r) = (maxDepth(root.left), maxDepth(root.right))
      (l max r) + 1
    }

  class TreeNode(var _value: Int = 0) {
    var value: Int = _value
    var left: TreeNode = null
    var right: TreeNode = null
  }
}
